src/main/scala/uk/gov/nationalarchives/fileformat/FFIDExtractor.scala

```diff
@@ -0,0 +1,56 @@
+package uk.gov.nationalarchives.fileformat
+
+import java.io.File
+import java.util.UUID
+
+import com.github.tototoshi.csv.CSVReader
+import com.typesafe.config.Config
+import graphql.codegen.types.{FFIDMetadataInput, FFIDMetadataInputMatches}
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse
+import uk.gov.nationalarchives.aws.utils.SQSUtils
+import uk.gov.nationalarchives.fileformat.FFIDExtractor._
+import scala.sys.process._
+import io.circe.syntax._
+
+import scala.util.Try
+
+class FFIDExtractor(sqsUtils: SQSUtils, config: Config) {
+  val sendMessage: String => SendMessageResponse = sqsUtils.send(config.getString("sqs.queue.output"), _)
+
+  def ffidFile(file: FFIDFile): Either[String, FFIDMetadataInput] = {
+    Try {
+      val efsRootLocation = config.getString("efs.root.location")
+      val command = s"$efsRootLocation/${config.getString("command")}"
+      val droidVersion = s"$command -v".!!.split("\n")(1)
+      val signatureOutput = s"$command -x".!!.split("\n")
+      val containerSignatureVersion = signatureOutput(1).split(" ").last
+      val droidSignatureVersion = signatureOutput(2).split(" ").last
+      s"""$command -a  $efsRootLocation/${file.consignmentId}/${file.originalPath} -p $efsRootLocation/${file.fileId}.droid""".!!
+      s"$command -p $efsRootLocation/${file.fileId}.droid -E $efsRootLocation/${file.fileId}.csv".!!
+      val reader = CSVReader.open(new File(s"$efsRootLocation/${file.fileId}.csv"))
+      implicit class OptFunction(str: String) {
+        def toOpt: Option[String] = if (str.isEmpty) Option.empty else Some(str)
+      }
+      val matches: List[FFIDMetadataInputMatches] = reader.all.tail.filter(o => o.length > 1 && o(1).isEmpty)
+        .map(o => FFIDMetadataInputMatches(o(9).toOpt, o(5), o(14).toOpt))
```

Comment:
Could you extract variables or find some other way to explain what `o(1)`, `o(9)`, etc. represent? Otherwise you have to dig around the DROID documentation to understand this code.

Reply:
I've extracted them into variables. It's possible to see what they're for if you look at the attributes for the FFIDMetadataInputMatches class but it's a lot clearer with them extracted as variables.